ignite/metrics/gan/fid.py

```diff
@@ -0,0 +1,213 @@
+from typing import Callable, Optional, Sequence, Union
+
+import numpy as np
+import torch
+
+from ignite.metrics.metric import Metric, reinit__is_reduced, sync_all_reduce
+
+__all__ = ["FID", "InceptionExtractor"]
+
+
+def fid_score(
+    mu1: torch.Tensor, mu2: torch.Tensor, sigma1: torch.Tensor, sigma2: torch.Tensor, eps: float = 1e-6
+) -> float:
+
+    try:
+        import scipy
+    except ImportError:
+        raise RuntimeError("fid_score requires scipy to be installed.")
+
+    mu1, mu2 = mu1.cpu(), mu2.cpu()
+    sigma1, sigma2 = sigma1.cpu(), sigma2.cpu()
+
+    diff = mu1 - mu2
+
+    # Product might be almost singular
+    covmean, _ = scipy.linalg.sqrtm(sigma1.mm(sigma2), disp=False)
+    if not np.isfinite(covmean).all():
+        offset = np.eye(sigma1.shape[0]) * eps
+        covmean = scipy.linalg.sqrtm((sigma1 + offset).dot(sigma2 + offset))
+
+    # Numerical error might give slight imaginary component
+    if np.iscomplexobj(covmean):
+        if not np.allclose(np.diagonal(covmean).imag, 0, atol=1e-3):
+            m = np.max(np.abs(covmean.imag))
+            raise ValueError("Imaginary component {}".format(m))
+        covmean = covmean.real
+
+    tr_covmean = np.trace(covmean)
+
+    return diff.dot(diff) + np.trace(sigma1) + np.trace(sigma2) - 2 * tr_covmean
+
+
+class InceptionExtractor:
+    def __init__(self) -> None:
+        try:
+            from torchvision import models
```

Comment:
Same than above, missing modules must be covered by tests

https://github.com/pytorch/ignite/blob/86cec1165441a945c93c938c23544ac52a17a170/tests/ignite/contrib/metrics/test_cohen_kappa.py#L23